aspnetcore/fundamentals/websockets.md

```diff
@@ -0,0 +1,98 @@
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/websockets/sample). See the [Next Steps](#next-steps) section for more information.
+
+
+## Prerequisites
+
+* ASP.NET Core 1.1 (does not run on 1.0)
+* Any OS that ASP.NET Core runs on:
+  
+  * Windows 7 / Windows Server 2008 and later
+  * Linux
+  * macOS
+
+   Exception: If your app runs on Windows with IIS, or with WebListener, you must use Windows 8 / Windows Server 2012 or later
+
+* For supported browsers, see http://caniuse.com/#feat=websockets.
+
+## When to use it
+
+Use WebSockets when you need to work directly with a socket connection. A typical reason is a need for best possible performance for real-time games.
+
+[ASP.NET SignalR](https://docs.microsoft.com/aspnet/signalr/overview/getting-started/introduction-to-signalr) provides a richer application model for real-time functionality, but SignalR does not run on ASP.NET Core. A Core version of SignalR is under development; to follow its progress, see the [GitHub repository for SignalR Core](https://github.com/aspnet/SignalR).
+
+If you don't want to wait for SignalR Core, you can use WebSockets directly now. But you might have to develop features that SignalR would provide, such as these:
+
+* Support for a broader range of browser versions by using automatic fallback to alternative transport methods.
+* Automatic reconnection when a connection drops.
+* Support for clients calling methods on the server or vice versa.
+* Support for scaling to multiple servers.
+
+## How to use it
+
+### Install the package
+
+Install [Microsoft.AspNetCore.WebSockets](https://www.nuget.org/packages/Microsoft.AspNetCore.WebSockets/).
+
+### Configure the middleware
+
+Add the WebSockets middleware in the `Configure` method of the `Startup` class.
+
+[!code-csharp[](websockets/sample/Startup.cs?name=UseWebSockets)]
+
+The following settings can be configured:
+
+* `KeepAliveInterval` - How frequently to send "ping" frames to the client, to ensure proxies keep the connection open.
+* `ReceiveBufferSize` - The size of the buffer used to receive data. Only a very advanced user would need to change this, for performance tuning based on the size of their data.
+
+[!code-csharp[](websockets/sample/Startup.cs?name=UseWebSocketsOptions)]
+
+### Accept a WebSocket request
+
+Somewhere later in the request life cycle (later in the `Configure` method or in an MVC action, for example) check if it's a WebSocket request and accept the WebSocket request.
+
+This example is from later in the `Configure` method.
+
+[!code-csharp[](websockets/sample/Startup.cs?name=AcceptWebSocket&highlight=7)]
```

Comment:
> How would it look to show the entire Configure method?

Right now it has both non-options and options code with if/endif around them so I can use them as snippets.  I could duplicate the method instead of just the app.Use part. What do you think?